### ISSUE TITLE
Inversion nom de voie / nom de commune (à cause de "Cédex" ?)

### ISSUE THREAD

**Issue by cquest** [addok/addok#101]
route de chalautre 77488 provins cedex
retourne:
Route de Provins 77160 Chalautre-la-Petite (77073_0230_c75c47 | 0.5096090909090909)
Route de Chalautre la Petite 77160 Provins (77379_XXXX_f32443 | 0.5025925133689839)
Rue de Provins 77160 Chalautre-la-Petite (77073_0240_5e33ef | 0.4463782531194295)

En retirant le "cedex", la bonne route est trouvée même en conservant le code postal incorrect (77488 au lieu de 77160)

**Comment by yohanboniface** [comment]
et avec le cedex MAIS le bon code postal?

**Comment by cquest** [comment]
ça ne va pas mieux... c'est vraiment CEDEX qui coince... un traitement spécifique ne ferait pas de mal, c'est un "mot" qui n'apparait nulle part dans la BAN.

D'ailleur essaye FREQUENCY cedex -> addok shell plante avec un index out of range

**Comment by yohanboniface** [comment]
`Cedex` est supprimé au moment de la recherche, là le problème est au niveau de la comparaison de chaîne. Jusqu'ici je me suis refusé à faire aucun traitement spécifique à ce niveau, parce que c'est justement le garde-fou. Mais à réfléchir.

Je regarde pour le bug dans FREQUENCY.

**Comment by frodrigo** [comment]
Cedex et son code postal pourait être supprimé dans un plugin spécifique France

**Comment by frodrigo** [comment]
'CEDEX' n'est probablement pas retiré au bon moment, il influe sur le résultat:

http://api-adresse.data.gouv.fr/search/?autocomplete=0&q=95153+TAVERNY+CEDEX&limit=1
=> Rue de Taverny 95130 Franconville

http://api-adresse.data.gouv.fr/search/?autocomplete=0&q=95153+TAVERNY&limit=1
=> 95150 Taverny

**Comment by cquest** [comment]
Je pense qu'il est retiré dans la première passe (recherche des mots dans l'index), mais pas dans la seconde (tri des candidats pour sélection).

**Comment by yohanboniface** [comment]
> mais pas dans la seconde (tri des candidats pour sélection)

Oui, mais c'est volontaire. Ce qui veut pas dire qu'il faut pas se reposer la question. Le principe, en très gros, est: on fait plein de magie pour trouver des candidats, puis on compare ces candidats à la *recherche initiale stricte*, justement parce que c'est ce qui sert de garde-fou, pour pas que les algos internes sortent des résultats sans queue ni tête.

**Comment by frodrigo** [comment]
Il faudrait peut-être la même chose mais à deux niveaux ?
- les changements que l'on est "sûr" de vouloir appliquer (cedex, cs, pb),
- ceux un peu trop magique avec le même comportement qu'actuellement.
